### ISSUE TITLE
MidiFile::mergeTracks needs to fix track field in all events that are moved

### ISSUE THREAD

**Issue by beldenfox** [craigsapp/midifile#54]
As `mergeTracks` moves the event lists upward to get rid of `aTrack2` it doesn't update the `track` field of the events being moved. The loop is

	for (int i=aTrack2; i<length-1; i++) {
		m_events[i] = m_events[i+1];
	}

After this loop all the events in `m_events[aTrack2]` on up will have `track` fields with their old track indices instead of their new ones.

**Comment by craigsapp** [comment]
Correct again!  Obviously I have not used `MidiFile::mergeTracks` very often...

I updated the above code snippet to:

```cpp
   for (int i=aTrack2; i<length-1; i++) {
      m_events[i] = m_events[i+1];
      for (int j=0; j<(int)m_events[i]->size(); j++) {
         (*m_events[i])[j].track = i;
      }
   }
```

Which updates the track field  in the individual events of the moved tracks.

In general I do not look at the `MidiEvent::track` variable when the file is in type-1 mode (multi-track).  When the file is in type-0 mode (single-track), it is often useful to know what the original multi-track track number was.  So previously the track information will be incorrect after a `mergeTracks` operation followed by a `joinTracks`.  Hopefully it is now better...  It might be best to do the track assignment in `joinTracks` before the tracks are merged to guarantee that they are correct, but this should be good enough for now.  Currently a library user could assign whatever they want as a track number (which might be another way to move tracks around by giving new track numbers, then doing joinTracks followed by splitTracks.

Fixed with commit https://github.com/craigsapp/midifile/commit/66c471ae8a2c5b115cfd52078556bb8e0b05f24c
